cmake/macros/macro_check_compiler_setup.cmake

```diff
@@ -27,10 +27,16 @@
 
 MACRO(CHECK_COMPILER_SETUP _compiler_flags_unstr _linker_flags_unstr _var)
   #
+  # Strip -Wl,--as-needed to work around a serious regression in ld.bfd.
+  # See https://github.com/dealii/dealii/issues/3686
+  #
```

Comment:
the issue is long, maybe add (reprhase):

Presumably, a deadlock in the linker is triggered by (a) our choice of program int main(){return 0;} and (b) -Wl,-as-needed in combination with (c) a ton of libraries.